src/app/workflow-editor/workflow-editor.component.ts

```diff
@@ -126,6 +130,34 @@ export class WorkflowEditorComponent extends AppEditorBase implements OnDestroy,
         }
     }
 
+    protected onFirstModelCreation(): void {
+        this.updateService.update
+            .filter((data: {id: string, app: any}) => {
+                let filterFn = (step) => false;
+                if (this.tabData.dataSource === "local") {
+                    filterFn = () => this.codeEditorContent.value.indexOf("run: " + data.id) > -1;
+                } else if (data.id) {
+                    filterFn = (step) => AppHelper.getRevisionlessID(step.run.customProps["sbg:id"] || "") === AppHelper.getRevisionlessID(data.id);
+                }
+                return this.dataModel.steps.filter(filterFn).length > 0;
+            })
+            .subscribeTracked(this, (data: {id: string, app: any}) => {
+                if (this.tabData.dataSource === "local") {
+                    const steps = this.dataModel.steps.filter(step => this.codeEditorContent.value.indexOf("run: " + data.id) > -1);
+                    steps.forEach(step => step.setRunProcess(data.app));
+                    this.syncModelAndCode(false).then(() => {
+                        this.resolveToModel(this.codeEditorContent.value);
```

Comment:
There shouldn't be a reason to call `this.resolveToModel` if `step.setRunProcess` is called, because that should have taken care of the update. This way, a new model is created for no reason.